enterprise_subsidy/apps/subsidy/api.py

```diff
@@ -38,3 +40,43 @@ def get_or_create_learner_credit_subsidy(
         defaults=subsidy_defaults,
     )
     return (subsidy, created)
+
+
+def can_redeem(subsidy, learner_id, content_key):
+    """
+    Determines if the given learner can redeem against the
+    provided subsidy for the given content_key.
+
+    Params:
+      subsidy: The Subsidy record against which redeemability is queried.
+      learner_id: (int) Primary (edX LMS) identifier of the learner for whom 
+        the redeemability query is being made.
+      content_key: (string) The content key of content for which
+        the redeemability query is being made.
+
+    Returns:
+      3-tuple of (
+        boolean: whether the user can redeem,
+        int: price of content,
+        Transaction: existing redemption/transaction
+      )
+    """
+    existing_transaction = subsidy.get_redemption(learner_id, content_key)
+    if existing_transaction and not _existing_reversal(existing_transaction):
+        is_redeemable = False
+        price_for_content = subsidy.price_for_content(content_key)
+    else:
+        is_redeemable, price_for_content = subsidy.is_redeemable(content_key)
+
+    return (is_redeemable, price_for_content, existing_transaction)
+
+
+def _existing_reversal(transaction):
```

Comment:
[curious/clarification] Very edge case-y, but can a transaction's reversal ever be reversed itself (i.e., "undo" a refund), where it _would_ count towards a prior redemption? If so, does this method take this into account? Probably something we can defer on.

Reply:
I don't think the models support that today.  To undo a refund, we'd probably have to just make a new redemption/create a new transaction.

Reply:
👍 Cool. That's probably a good thing the models don't support that today; less complexity to deal with 😄